docs/examples/userguide/extension_types/dataclass.pyx

```diff
@@ -0,0 +1,15 @@
+cimport cython
+
+@cython.dataclasses.dataclass
+cdef class MyDataclass:
+    # fields can be declared using annotations
+    a: cython.int = 0
+    b: double = cython.dataclasses.field(default_factory = lambda: 10, repr=False)
+
+    # fields can also be declared using `cdef`:
+    cdef str c
+    c = "hello"  # assignment of default value on a separate line
+
+    # cython equivalents to InitVar and typing.ClassVar also work
+    d: cython.dataclasses.InitVar[double] = 5
+    e: cython.typing.ClassVar[list] = []
```

Comment:
I definitely agree that making our own copies of these is awkward. It needs some mechanism to track these attributes (i.e. how do we know that a name `ClassVar` comes from the dataclass module) which I don't think currently exists. It might be useful infrastructure though for dealing with things like the `typing` module.

> Basically, if you use this feature in .pyx files, then the fallbacks aren't really needed for the compile time declarations. If you use it in .py files, then you'll need a suitable Python version anyway, and still don't need the fallbacks.

I can see the value in not trying to provide fallbacks. For something like:

```
# some_py_file.py
import cython
import dataclasses

@cython.cclass
@cython.dataclass
class C:
  d: dataclasses.InitVar[double] = 5
```

the `import dataclasses` line would still be executable and still fail on older versions of Python, even if Cython interpreted the rest correctly. I don't really know how to deal with that.

Reply:
Right - with the current version of changes based on https://github.com/cython/cython/pull/3910 we don't need these any more.